pyomo/neos/tests/test_neos.py

```diff
@@ -90,6 +88,177 @@ def test_kestrel_plugin(self):
         self.assertEqual(results.solver[0].status, pyo.SolverStatus.ok)
         self.assertAlmostEqual(pyo.value(m.x), 0.5)
 
+    def test_doc(self):
+        kestrel = kestrelAMPL()
+        tmp = [tuple(name.split(':')) for name in kestrel.solvers()]
+        amplsolvers = set(v[0].lower() for v in tmp if v[1]=='AMPL')
+
+        doc = pyomo.neos.doc
+        dockeys = set(doc.keys())
+
+        self.assertEqual(amplsolvers, dockeys)
+
+        #gamssolvers = set(v[0].lower() for v in tmp if v[1]=='GAMS')
+        #missing = gamssolvers - amplsolvers
+        #print("HERE", missing)
+        #self.assertEqual(len(missing) == 0)
+
+
+@unittest.category('neos')
+@unittest.skipIf(not neos_available, "Cannot make connection to NEOS server")
+class TestSolvers_script_min(unittest.TestCase):
+
+    def _model(self):
+        m = pyo.ConcreteModel()
+        m.x = pyo.Var(bounds=(0,1), initialize=0)
+        m.c = pyo.Constraint(expr=m.x <= 0.5)
+        m.obj = pyo.Objective(expr=-2*m.x, sense=pyo.minimize)
+        return m
+
+    def _run(self, opt):
+        m = self._model()
+        solver_manager = pyo.SolverManagerFactory('neos')
+        results = solver_manager.solve(m, opt=opt)
+
+        self.assertEqual(results.solver[0].status, pyo.SolverStatus.ok)
+        self.assertAlmostEqual(pyo.value(m.x), 0.5)
+
+    def test_bonmin(self):
+        self._run('bonmin')
+
+    def test_cbc(self):
+        self._run('cbc')
+
+    def test_conopt(self):
+        self._run('conopt')
+
+    def test_couenne(self):
+        self._run('couenne')
+
+    def test_cplex(self):
+        self._run('cplex')
+
+    def test_filmint(self):
+        self._run('filmint')
+
+    def test_filter(self):
+        self._run('filter')
+
+    def test_ipopt(self):
+        self._run('ipopt')
+
+    def test_knitro(self):
+        self._run('knitro')
+
+    # This solver does not return
+    # a solution when all variables are
+    # projected to be at their bounds.
+    #def test_lbfgsb(self):
+    #    self._run('l-bfgs-b')
+
+    def test_lancelot(self):
+        self._run('lancelot')
+
+    def test_loqo(self):
+        self._run('loqo')
+
+    def test_minlp(self):
+        self._run('minlp')
+
+    def test_minos(self):
+        self._run('minos')
+
+    # This solver doesn't generate a solution file
+    # when the presolver finds the solution
+    #def test_minto(self):
+    #    self._run('minto')
+
+    def test_mosek(self):
+        self._run('mosek')
+
+    # This solver doesn't give the same answer for
+    # min f(x) and max -f(x)
+    #def test_ooqp(self):
+    #    self._run('ooqp')
+
+    # The simple tests aren't complementarity 
+    # problems
+    #def test_path(self):
+    #    self._run('path')
+
+    def test_snopt(self):
+        self._run('snopt')
+
+    # This test only handles minimization problems
+    #def test_raposa(self):
+    #    self._run('raposa')
```

Comment:
With a class level attribute that indicates the sense, we could skip this test for maximization cases.